server/utils/firestore.ts

```diff
@@ -0,0 +1,35 @@
+import { initializeApp, getApps } from 'firebase-admin/app'
+import { getFirestore } from 'firebase-admin/firestore'
+import type { App } from 'firebase-admin/app'
+import type { Firestore } from 'firebase-admin/firestore'
+
+let app: App | undefined
+let db: Firestore | undefined
+
+function initializeFirebase() {
+  if (getApps().length === 0) {
+    app = initializeApp()
+  }
+  else {
+    app = getApps()[0]
+  }
+
+  if (!app) {
+    throw new Error('Failed to initialize Firebase app')
```

Comment:
Prefer using `createError`
```suggestion
    throw createError({ statusMessage: 'Failed to initialize Firebase app' })
```